### DISCUSSION TITLE
🆘 GraphQL Query on arrays

### DISCUSSION THREAD

**Discussion by iFrogHop2Worlds** [tinacms/tinacms#4153] [General]
Never used GraphQL before trying tina so I am sorry if this is not really tina related. I am trying to grab an object from an array of objects within a collection. Not really sure how to structure the query. The way I have things setup right now is such that a collection is a category and within a category you have an array of products. Each product is an object. Id like to grab that object.

this is what I am currently doing and works for this simple use-case fine but if anyone can help me understand how to use resolver to fix this with graphql properly would appreciate it.

My current code.  My naming conventions were horrible here as I changed things several times on the project just really focused on learning tina and I guess graphQL now. fileName is the .MD file obviously category items is an array of  the objects I would like to isolate.
```
    const q = await client.queries.product({
      relativePath: fileName, 
      query: `fragment ProductParts on Product {
        category_items 
      }`
    })

    let product = q.data.product.category_items.filter(item => {
      //console.log(title)
      return item.title === title
    })  
```

**Comment by jeffsee55** [comment]
With Tina, the `client.queries` api is a set of pre-generated queries, you can see exactly what they are in `tina/__generated__/queries`. But you won't be able to append fragments to it with that you've got there, however, in this case the response for `client.queries.product` should already include the `category_items` field.
